### ISSUE TITLE
FR: GET API with query params

### ISSUE THREAD

**Issue by yeleleo** [lyqht/mini-qr#50]
I wanted to thank you for such a great app! It is really great!

Wanted to check if you are planning to make any API protocol to generate QR codes on a fly via `<img src="">` on a webpage (dynamically changing parameters)

Something like:

```
https://localhost/api?data=https://github.com
&error=H
&logo=https://avatars.githubusercontent.com/u/1327411?s=80&v=4
&background_color=
&dots_color=
&corners_square_color=
&corners_dot_color=
&width=
&height=
&margin=
&image_margin=
&border_radius=
&dots_type=
&corners_square_type=
&corners_dot_type=
```

It will be highly appreciated!

**Comment by lyqht** [comment]
Thanks for raising this issue and Im glad you like the app!

I was thinking of something this before, but introducing APIs would lead to the creation of a db and a server, which would then require me to spend $. And if that's the case, I'll have to paywall that feature to sustain the costs.

What are your thoughts about this? Would u still be keen in this feature if you have to pay for it?

**Comment by lyqht** [comment]
I can start off this feature by supporting search query params to auto populate the qr code design, but I'm not too sure of the part to support GET request of a URL yet.

**Comment by yeleleo** [comment]
Good to hear that that you've thought about it!

There are tons on commercial QR-code cloud APIs but there are almost no API solutions which can be installed on own server for internal needs (non-public use) via localhost access. 

It can be build w'o database, just generate image on GET with querystring and cache the final image, no need to store the file or request.

To monetise it I would suggest another way - make a docker image and distribute it with Marketplaces as one on Digital Ocean – https://marketplace.digitalocean.com/

**Comment by Mapaor** [comment]
Maybe I'm missing something but wouldn't it be as simple as creating a mirror of this repo as a nextjs app deployed to Vercel that has an api route that simply returns the svg code?

**Comment by lyqht** [comment]
@Mapaor yes it's possible to host it however you want (even through a fullstack solution like Next.js when you don't even need the frontend for it). 

However, for production grade reliability and scalability, APIs should be able to handle high request rates and set appropriate limits. This is not something that will be handled by default by simply spinning up a random free Next.js instance by Vercel.

But of course if your use case is low traffic, by all means you can do what many others are already doing with the docker image.

**Comment by jorgedanisc** [comment]
Not sure how the app is being deployed, but this most likely could be achieved from the client-side. As @lyqht said, wouldn't `supporting search query params` be sufficient here for the end goal of loading presets from an url?

Since presets are purely a JSON, this could be loaded via base64 or else through the search params without extra costs

**Comment by KeithBrown39423** [comment]
As someone who as attempted to make a qrcode api before, POST requests would be better mainly because of the data payload supporting so many different options. As for the API itself and hosting, Google Cloud Run is also an option that would scale with use-case, not costing anything unless calls are made

**Comment by yeleleo** [comment]
POST api generates the main blocker - it is not possible to use inline, specifically on email layouts. 
Initial request was about GET + querry, so for post better to create another ticket.

**Comment by cdesign89** [comment]
最近在做一个项目需要批量制作一些二维码，一番搜寻后发现这个项目，做得很漂亮，功能也很好！！
一开始我是想通过docker部署在NAS上，再通过api来实现，这样能实现不用产生一堆二维码文件再重新导入，但后来一直都找不到比较合适项目，最后只能用了quickchart的api生成。
很希望作者能尝试加入api的功能😄~！再次感谢您制作了如此精美好用的项目。
